homeassistant/components/sonarr/sensor.py

```diff
@@ -72,198 +61,481 @@
-        vol.Optional(CONF_INCLUDED, default=[]): cv.ensure_list,
-        vol.Optional(CONF_MONITORED_CONDITIONS, default=["upcoming"]): vol.All(
-            cv.ensure_list, [vol.In(list(SENSOR_TYPES))]
-        ),
-        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
-        vol.Optional(CONF_SSL, default=False): cv.boolean,
-        vol.Optional(CONF_UNIT, default=DEFAULT_UNIT): vol.In(BYTE_SIZES),
-        vol.Optional(CONF_URLBASE, default=DEFAULT_URLBASE): cv.string,
-    }
+
+DEFAULT_URLBASE = ""
+DEFAULT_DAYS = "1"
+DEFAULT_UNIT = DATA_GIGABYTES
+
+PLATFORM_SCHEMA = vol.All(
+    cv.deprecated(CONF_INCLUDED, invalidation_version="0.112"),
+    cv.deprecated(CONF_MONITORED_CONDITIONS, invalidation_version="0.112"),
+    cv.deprecated(CONF_UNIT, invalidation_version="0.112"),
+    PLATFORM_SCHEMA.extend(
+        {
+            vol.Required(CONF_API_KEY): cv.string,
+            vol.Optional(CONF_DAYS, default=DEFAULT_DAYS): cv.string,
+            vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
+            vol.Optional(CONF_INCLUDED, default=[]): cv.ensure_list,
+            vol.Optional(CONF_MONITORED_CONDITIONS, default=[]): cv.ensure_list,
+            vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+            vol.Optional(CONF_SSL, default=DEFAULT_SSL): cv.boolean,
+            vol.Optional(CONF_UNIT, default=DEFAULT_UNIT): vol.In(BYTE_SIZES),
+            vol.Optional(CONF_URLBASE, default=DEFAULT_URLBASE): cv.string,
+        }
+    ),
 )
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the Sonarr platform."""
-    conditions = config.get(CONF_MONITORED_CONDITIONS)
-    add_entities([SonarrSensor(config, sensor) for sensor in conditions], True)
+async def async_setup_platform(
+    hass: HomeAssistantType,
+    config: ConfigType,
+    async_add_entities: Callable[[List[Entity], bool], None],
+    discovery_info: Any = None,
+) -> None:
+    """Import the platform into a config entry."""
+    if len(hass.config_entries.async_entries(DOMAIN)) > 0:
+        return True
+
+    config[CONF_BASE_PATH] = f"{config[CONF_URLBASE]}{DEFAULT_BASE_PATH}"
+    config[CONF_UPCOMING_DAYS] = int(config[CONF_DAYS])
+    config[CONF_VERIFY_SSL] = False
+
+    del config[CONF_DAYS]
+    del config[CONF_INCLUDED]
+    del config[CONF_MONITORED_CONDITIONS]
+    del config[CONF_URLBASE]
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=config
+        )
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType,
+    entry: ConfigEntry,
+    async_add_entities: Callable[[List[Entity], bool], None],
+) -> None:
+    """Set up Sonarr sensors based on a config entry."""
+    options = entry.options
+    sonarr = hass.data[DOMAIN][entry.entry_id][DATA_SONARR]
+    unique_id = entry.unique_id
+
+    if unique_id is None:
+        unique_id = entry.entry_id
```

Comment:
Should this be done during config entry creation? Otherwise you could end up messing with unique IDs if we accidentally update this code.

Reply:
so would i just set the unique id to entry_id in setup and use unique_id here? or via hass.data?

Reply:
its unlikely that there will ever be a true unique_id from sonarr itself

Reply:
went a different direction and just use entry_id always since the future proofing was not really needed.